app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -308,38 +295,36 @@ public Block createRevokeBlock(String owner, String contact, String publicKey, S
      * @param contact   of whom is the information
      * @param publicKey public key you want to store
      * @param iban      IBAN number to store in this block
-     * @param revoke    whether to revoke?
+     * @param blockType type of the block
      * @return the newly created block
      * @throws Exception when the hashing algorithm is not available
      */
-    private Block createBlock(String owner, String contact, String publicKey, String iban, boolean
-            revoke) throws Exception {
-        Block latest = getLatestBlock(owner);
+    private Block createBlock(User owner, User contact, String publicKey, String iban,
+                              BlockType blockType) throws HashException {
+        Block latest = getLatestBlock(owner.getName());
+        if (latest == null)
+            throw new IllegalArgumentException("No genesis found for user " + owner);
         String previousBlockHash = latest.getOwnHash();
 
         // always link to genesis of contact blocks
         String contactBlockHash;
-        if (owner.equals(contact))
+        if (owner.equals(contact)) {
             contactBlockHash = NA;
-        else
-            contactBlockHash = getBlocks(contact).get(0).getOwnHash();
+        } else {
+            contactBlockHash = getBlocks(contact.getName()).get(0).getOwnHash();
```

Comment:
Also outside the scope of this PR; this information was not available at the time of this PR.